lib/internal/bootstrap/cache.js

```diff
@@ -9,22 +9,49 @@ const {
   NativeModule, internalBinding
 } = require('internal/bootstrap/loaders');
 
+function getCodeCache(id) {
+  const cached = NativeModule.getCached(id);
+  if (cached && (cached.loaded || cached.loading)) {
+    const script = cached.script;
+    return script.createCachedData();
+  }
+
+  // The script has not been compiled and run
+  NativeModule.require(id);
+  return getCodeCache(id);
+}
+
+const depsModule = Object.keys(NativeModule._source).filter(
+  (key) => NativeModule.isDepsModule(key) || key.startsWith('internal/deps')
+);
+
+// Modules with source code compiled in js2c that
```

Comment:
I'm not sure I understand this comment. Why can't these scripts be cached?

Reply:
- `config` is a JSON string, it's not going to be wrapped
- `sys` is deprecated and is redirected from `util` (executing it would trigger a deprecation warning)
- `internal/test/binding` triggers a warning as well (also, it's not supposed to be loaded by users)
- `internal/v8_prof_polyfill` is not supposed to be required directly (see the comments there)
- `internal/v8_prof_processor` is an actual script (instead of a module) that is supposed to write a log file

The files in v8/tools are not really Node.js modules. They are mostly concated and run by `internal/v8_prof_processor` since they exports stuff to the global space an rely on that.

Reply:
I see. Thanks for the explanation.